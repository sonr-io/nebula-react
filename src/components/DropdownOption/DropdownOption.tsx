import React from 'react';
import { DropdownOptionProps } from '../../types/dropdownOptionProps';
import { NebulaIcon } from '../Icons/NebulaIcon';

const cx = require('classnames');

export const DropdownOption: React.FC<DropdownOptionProps> = ({
    title,
    leftIconName,
    leftIconType = 'filled',
    rightIconName,
    rightIconType = 'filled',
}) => {
    const dropDownClass = cx()

    return (
        <div className='relative cursor-pointer'>
            {leftIconName && <NebulaIcon iconName={leftIconName} iconType={leftIconType} />}
            {title}
            {rightIconName && <NebulaIcon iconName={rightIconName} iconType={rightIconType} />}
        </div>
    )
}
