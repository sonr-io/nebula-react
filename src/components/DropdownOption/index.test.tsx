import { render } from '@testing-library/react'
import { DropdownOption } from "./DropdownOption"

jest.mock("../../assets/outline/Document.svg", () => "document-svg")
jest.mock("../../assets/filled/Document.svg", () => "document-svg")
jest.mock("../../assets/duotone/Document.svg", () => "document-svg")

beforeEach(() => {
    jest.resetModules();
});  

describe('DropdownOption component', () => {
    test('should render', () => {
        const { asFragment } = render(
            <DropdownOption  
                title='Option 1'
                leftIconName='Document'
                leftIconType='outline'
            />
        )

        expect(asFragment()).toMatchSnapshot()
    })
})