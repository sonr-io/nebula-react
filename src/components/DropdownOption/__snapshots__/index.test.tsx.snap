// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownOption component should render 1`] = `
<DocumentFragment>
  <div
    class="flex items-center cursor-pointer px-8 h-10 text-gray-800 hover:text-white hover:bg-primary-light hover:fill-white"
  >
    <document-svg
      classname="w-4 h-4 inline w-4 h-4 w-6 h-6 mr-2.5"
      data-testid="nebula-icon"
    />
    <span
      class="text-custom-xs font-extrabold"
    >
      Option 1
    </span>
  </div>
</DocumentFragment>
`;
